### ISSUE TITLE
Resize tf_half_pixel_for_nn

### ISSUE THREAD

**Issue by kraiskil** [onnx/onnx#3924]
# Bug Report

### Describe the bug
The `Resize` operator had a `coordinate_transformation_mode` attribute value `tf_half_pixel_for_nn` introduced in opset version 11, but removed in version 13.

Yet there still exists the `onnx/backend/test/data/node/test_resize_downsample_sizes_nearest_tf_half_pixel_for_nn/` backend test with the opset version 13 and that value set to the attribute.

### Expected behavior
Either the quoted test should be removed or the option is re-introduced in the documentation for `Resize`.

Possibly the test could remain, if the opset version is reduced back to 11 for it - but it looks like the strategy for ONNX tests is to remove old opset  version tests from the HEAD of the repo, so this doesn't sound like the correct solution

**Comment by askhade** [comment]
IMO we should remove the test for the very same reason you stated: "but it looks like the strategy for ONNX tests is to remove old opset version tests from the HEAD of the repo"

**Comment by garymm** [comment]
It's surprising to me that the checker doesn't fail when the model has an invalid attribute value. Do we have an issue tracking enforcing attribute values are valid? Shall I open one?

**Comment by garymm** [comment]
Not sure we should just delete this test. I tried setting the opset version to 11 to make it valid and it uncovered a bug: #3959. Will discuss and then come back to this.

**Comment by askhade** [comment]
Checker does not have this check because we dont have a list of valid values in code. This info is in the spec. So, the checker only knows data of type string is valid but it does not have a way to know which strings are valid.

**Comment by askhade** [comment]
@garymm : The issue you observed existed in opset 11. The fix was added in opset 13. 
PR which fixed the issue: https://github.com/onnx/onnx/pull/3026

**Comment by kraiskil** [comment]
Well, this was a bit disappointing resolution. I missed the notification on the pull request, so this comes too late. Sorry about that.

The commit above now includes now old (deprecated?) Opset version tests in the backed unit tests. Does this mean the above mentioned strategy has changed? Or is this more of an exception?

Somewhat unrelated, this change to the `test_resize_downsample_sizes_nearest_tf_half_pixel_for_nn`  where the `sizes` tensor nolonger is constant makes this test unusable (at least without real significant changes in the infra running the tests) for my backend where dynamic memory allocation is not allowed. I know that I'm in a minority here, but I'm hoping this does not become a trend in the backend tests :)

**Comment by garymm** [comment]
@kraiskil I took on the task of fixing this and I think there is utility in having test coverage for older op sets, so I chose to make the test valid rather than delete it. I think it'd be reasonable to open an issue along the lines of `document policy about backend tests for older op set versions`.

Feel free to send a PR to make the `sizes` tensor constant if that makes your life easier without reducing the utility of the test.
